test/components/date-picker.spec.tsx

```diff
@@ -28,396 +34,326 @@ describe('The DatePicker Component', () => {
+        datePicker.changeDate('2017/02/01');
 
         await waitForDom(() => expect(onChange).to.have.been.calledWithMatch({value: FEBRUARY_FIRST}));
     });
 
     it('should use a provided placeholder', async () => {
-        const {select, waitForDom} = clientRenderer.render(<DatePicker placeholder="mm/dd/yyyy" />);
+        const {driver: datePicker, waitForDom} = clientRenderer.render(<DatePicker placeholder="mm/dd/yyyy" />)
+            .withDriver(DatePickerTestDriver);
 
-        await waitForDom(() => expect(select(datePickerInputId)).to.have.attribute('placeholder', 'mm/dd/yyyy'));
+        await waitForDom(() => expect(datePicker.input).to.have.attribute('placeholder', 'mm/dd/yyyy'));
     });
 
     it('should show and hide the dropdown when the input is clicked', async () => {
-        const {select, waitForDom} = clientRenderer.render(<DatePicker />);
-        const datePickerInput = select(datePickerInputId);
-
-        await waitForDom(() => expect(bodySelect(datePickerDropdownId)).to.be.absent());
-
-        simulate.mouseDown(datePickerInput);
-
-        await waitForDom(() => expect(bodySelect(datePickerDropdownId)).to.be.present());
-
-        simulate.mouseDown(datePickerInput);
-
-        await waitForDom(() => expect(bodySelect(datePickerDropdownId)).to.be.absent());
-    });
-
-    it('should show and hide the dropdown when the calendar icon is clicked', async () => {
-        const {select, waitForDom} = clientRenderer.render(<DatePicker />);
-        const calendarIcon = select('CALENDAR_ICON');
+        const {driver: datePicker, waitForDom} = clientRenderer.render(<DatePicker />)
+            .withDriver(DatePickerTestDriver);
 
-        await waitForDom(() => expect(bodySelect(datePickerDropdownId)).to.be.absent());
+        await waitForDom(() => expect(datePicker.isOpen()).to.be.false);
 
-        simulate.click(calendarIcon);
+        datePicker.clickOnDatePicker();
 
-        await waitForDom(() => expect(bodySelect(datePickerDropdownId)).to.be.present());
+        await waitForDom(() => expect(datePicker.isOpen()).to.be.true);
 
-        simulate.click(calendarIcon);
+        datePicker.clickOnDatePicker();
 
-        await waitForDom(() => expect(bodySelect(datePickerDropdownId)).to.be.absent());
+        await waitForDom(() => expect(datePicker.isOpen()).to.be.false);
     });
 
     it('should show and hide the dropdown when focused and openOnFocus is true', async () => {
-        const {select, waitForDom} = clientRenderer.render(<DatePicker openOnFocus />);
-        const datePickerInput = select(datePickerInputId);
+        const {driver: datePicker, waitForDom} = clientRenderer.render(<DatePicker openOnFocus />)
+            .withDriver(DatePickerTestDriver);
 
-        await waitForDom(() => expect(bodySelect(datePickerDropdownId)).to.be.absent());
+        await waitForDom(() => expect(datePicker.dropDown).to.be.absent());
 
-        simulate.focus(datePickerInput);
+        datePicker.input.focus();
```

Comment:
I would abstract the focus to the driver. So `datePicker.focus()`. Also, please remove the comment.